From dffbe31781326c55861374592c369d1225124b82 Mon Sep 17 00:00:00 2001
From: Wei-Chiu Chuang <weichiu@cloudera.com>
Date: Mon, 2 May 2016 06:44:21 -0700
Subject: [PATCH 1479/2748] HDFS-9629. Update the footer of Web UI to show
 year 2016 (Contributed by Xiao Chen and
 Vinayakumar B)

(cherry picked from commit ddcd6954d03d3caadc8b69f7b886a3a17c3c9d0c)
(cherry picked from commit 8cdfdbf815bf89eee3184ec7cd00e1924c1ea316)

 Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt
	hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/datanode/index.html
	hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.html
	hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.html
	hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/journal/index.html
	hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/secondary/status.html

(cherry picked from commit ea517acfb6eb414e85be32b21d6240301744a306)

Change-Id: Idad8f497ba30cee1b4ae3438b56a21413bf8badd
---
 hadoop-hdfs-project/hadoop-hdfs/pom.xml            |    4 ++++
 .../src/main/webapps/datanode/index.html           |    2 +-
 .../src/main/webapps/hdfs/dfshealth.html           |    2 +-
 .../src/main/webapps/hdfs/explorer.html            |    2 +-
 .../src/main/webapps/journal/index.html            |    2 +-
 .../src/main/webapps/secondary/status.html         |    2 +-
 hadoop-project/pom.xml                             |    3 +++
 7 files changed, 12 insertions(+), 5 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/pom.xml b/hadoop-hdfs-project/hadoop-hdfs/pom.xml
index b947d884..e7e6432 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/pom.xml
+++ b/hadoop-hdfs-project/hadoop-hdfs/pom.xml
@@ -419,6 +419,10 @@ http://maven.apache.org/xsd/maven-4.0.0.xsd">
                     <exclude name="**/proto-*-web.xml"/>
                   </fileset>
                 </copy>
+                <replace dir="${project.build.directory}/webapps" value="${release-year}">
+                  <include name="**/*.html"/>
+                  <replacetoken>{release-year-token}</replacetoken>
+                </replace>
               </target>
             </configuration>
           </execution>
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/datanode/index.html b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/datanode/index.html
index 04cb703..a88bc9b 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/datanode/index.html
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/datanode/index.html
@@ -47,7 +47,7 @@
 
 <div class="row">
   <hr />
-  <div class="col-xs-2"><p>Hadoop, 2014.</p></div>
+  <div class="col-xs-2"><p>Hadoop, {release-year-token}.</p></div>
 </div>
 </div>
 
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.html b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.html
index 267dcce..c2889b8 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.html
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.html
@@ -68,7 +68,7 @@
 
 <div class="row">
   <hr />
-  <div class="col-xs-2"><p>Hadoop, 2014.</p></div>
+  <div class="col-xs-2"><p>Hadoop, {release-year-token}.</p></div>
   <div class="col-xs-1 pull-right"><a style="color: #ddd" href="dfshealth.jsp">Legacy UI</a></div>
 </div>
 </div>
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.html b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.html
index 34ee3f7..47086de 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.html
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.html
@@ -99,7 +99,7 @@ <h4 class="modal-title" id="file-info-title">File information</h4>
 
       <div class="row">
         <hr />
-        <div class="col-xs-2"><p>Hadoop, 2014.</p></div>
+        <div class="col-xs-2"><p>Hadoop, {release-year-token}.</p></div>
       </div>
 
     </div>
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/journal/index.html b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/journal/index.html
index 5eabf64..72ab2cc 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/journal/index.html
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/journal/index.html
@@ -47,7 +47,7 @@
 
 <div class="row">
   <hr />
-  <div class="col-xs-2"><p>Hadoop, 2014.</p></div>
+  <div class="col-xs-2"><p>Hadoop, {release-year-token}.</p></div>
 </div>
 </div>
 
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/secondary/status.html b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/secondary/status.html
index 9542ab7..3db97df 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/secondary/status.html
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/secondary/status.html
@@ -53,7 +53,7 @@
 
 <div class="row">
   <hr />
-  <div class="col-xs-2"><p>Hadoop, 2014.</p></div>
+  <div class="col-xs-2"><p>Hadoop, {release-year-token}.</p></div>
 </div>
 </div>
 
diff --git a/hadoop-project/pom.xml b/hadoop-project/pom.xml
index 8238747..46c59c3 100644
--- a/hadoop-project/pom.xml
+++ b/hadoop-project/pom.xml
@@ -30,6 +30,9 @@
   <packaging>pom</packaging>
 
   <properties>
+    <!-- Set the Release year during release -->
+    <release-year>2016</release-year>
+
     <failIfNoTests>false</failIfNoTests>
     <maven.test.redirectTestOutputToFile>true</maven.test.redirectTestOutputToFile>
 
-- 
1.7.9.5

